✨   MS preflight checks to improve cluster stability (#8595)
kubernetes-sigs/cluster-api: docs/book/src/tasks/experimental-features/machineset-preflight-checks.md

# Experimental Feature: MachineSetPreflightChecks (alpha)

> The `MachineSetPreflightChecks` feature is introduced to provide additional safety while creating new Machines for a MachineSet.

Reviewer: ```suggestion
The `MachineSetPreflightChecks` feature can provide additional safety while creating new Machines for a MachineSet.
```

**Feature gate name**: `MachineSetPreflightChecks`

**Variable name to enable/disable the feature gate**: `EXP_MACHINE_SET_PREFLIGHT_CHECKS`

The following preflight checks are performed when the feature is enabled:
* ControlPlaneStable
* KubeadmVersion

> * KubernetesVersion

Reviewer: Is it possible to provide a user-oriented explanation about what is check is implemented/why it is important?

Also, what about adding a note about the possibility to skip preflight check by exception/on a MD or MS bases once the feature flag is enabled?

Reviewer 2: I assume the plan was to do this in https://github.com/kubernetes-sigs/cluster-api/issues/8717 as a follow-up

Author: Yup, there is a follow-up issue to improve the documentation as discussed here: https://github.com/kubernetes-sigs/cluster-api/pull/8595#discussion_r1200422174
